File: frontend/lib/models/user_role.dart
/// Énumération des rôles utilisateur
enum UserRole {
  technician,
  admin;

  /// Convertit une string en UserRole
  static UserRole fromString(String role) {
    return UserRole.values.firstWhere(
      (e) => e.name.toLowerCase() == role.toLowerCase(),
      orElse: () => UserRole.technician,
    );
  }

  /// Retourne le nom du rôle pour l'API
  String toApiString() {
    switch (this) {
      case UserRole.admin:
        return 'Admin';
      case UserRole.technician:
        return 'Technician';
    }
  }
}

File: backend/src/NovadisApi/Models/UserRole.cs
namespace NovadisApi.Models
{
    /// <summary>
    /// Énumération des rôles utilisateur
    /// </summary>
    public enum UserRole
    {
        Technician,
        Admin
    }

    /// <summary>
    /// Extensions pour le rôle utilisateur
    /// </summary>
    public static class UserRoleExtensions
    {
        public static UserRole FromString(string role)
        {
            return role?.ToLower() switch
            {
                "admin" => UserRole.Admin,
                "technician" => UserRole.Technician,
                "technicien" => UserRole.Technician,
                _ => UserRole.Technician
            };
        }

        public static string ToRoleString(this UserRole role)
        {
            return role switch
            {
                UserRole.Admin => "Admin",
                UserRole.Technician => "Technician",
                _ => "Technician"
            };
        }
    }
}
